fix(api): Perform grip point calculations depending on labware schema version

mjhuff opened the pull request:
Closes [EXEC-1803](https://opentrons.atlassian.net/browse/EXEC-1803)

<!--
Thanks for taking the time to open a Pull Request (PR)! Please make sure you've read the "Opening Pull Requests" section of our Contributing Guide:

https://github.com/Opentrons/opentrons/blob/edge/CONTRIBUTING.md#opening-pull-requests

GitHub provides robust markdown to format your PR. Links, diagrams, pictures, and videos along with text formatting make it possible to create a rich and informative PR. For more information on GitHub markdown, see:

https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax

To ensure your code is reviewed quickly and thoroughly, please fill out the sections below to the best of your ability!
-->

# Overview

See the ticket for background.

In order to ensure proper gripping behavior, we have to do a couple things:

- 0c14a0aee049d5fc474b63f85140038ab30b5e8f: Gate the calculation by labware definition version. We effectively use the exact same code prior to the https://github.com/Opentrons/opentrons/pull/19354 changes for schema 2. See the ticket. There are too many non-auditable definition changes, which is why we want to gate changes in the first place.

- f2e0af2650bc5226f70e36d0585fa631adb841db: Yep, turns out we had a bug in at least 8.4.1 onwards (and realistically, probably much earlier) that effectively assumes that the x/y values of `stackingOffsetWithLabware` must be 0. Both gripping labware and pipetting make use of these X/Y values, and modifying them to any non-zero value effectively creates a new offset that is problematic. If you don't feel like auditing code, throwing a robot on 8.4.1 and pushing some non-zero X/Y `stackingOffsetWithLabware` values (see below) and running the following protocol causes the issue:

Stacking values in `opentrons_flex_tiprack_lid`:
```
  "opentrons_flex_96_tiprack_1000ul": {
      "x": 10,
      "y": 0,
      "z": 14.25
    }
```
Protocol:
```
requirements = {"apiLevel": "2.25", "robotType": "Flex"}

from opentrons import protocol_api

def run(protocol: protocol_api.ProtocolContext) -> None:
    tip_rack = protocol.load_labware(
        "opentrons_flex_96_tiprack_1000ul", "D1", lid="opentrons_flex_tiprack_lid"
    )
    trash = protocol.load_trash_bin("A3")

    protocol.move_labware(tip_rack, "D2", use_gripper=True)
    protocol.move_lid(tip_rack, trash, use_gripper=True)
```

So the schema definition has been updated to reflect the fact that we only expect the z-value of this property to change. It's always been buggy, and we're unfortunately tied to its bugginess (until schema 3 that is).
<!--
Describe your PR at a high level. State acceptance criteria and how this PR fits into other work. Link issues, PRs, and other relevant resources.
-->

## Test Plan and Hands on Testing
- Ran the above protocol for schema 2 with the reverted definition changes, ensuring gripping works as expected.
<!--
Describe your testing of the PR. Emphasize testing not reflected in the code. Attach protocols, logs, screenshots and any other assets that support your testing.
-->

## Changelog
- The gripper correctly picks up and moves Flex tip rack lids.
<!--
List changes introduced by this PR considering future developers and the end user. Give careful thought and clear documentation to breaking changes.
-->

<!--
- What do you need from reviewers to feel confident this PR is ready to merge?
- Ask questions.
-->

## Risk assessment
low
<!--
- Indicate the level of attention this PR needs.
- Provide context to guide reviewers.
- Discuss trade-offs, coupling, and side effects.
- Look for the possibility, even if you think it's small, that your change may affect some other part of the system.
  - For instance, changing return tip behavior may also change the behavior of labware calibration.
- How do your unit tests and on hands on testing mitigate this PR's risks and the risk of future regressions?
- Especially in high risk PRs, explain how you know your testing is enough.
-->


[EXEC-1803]: https://opentrons.atlassian.net/browse/EXEC-1803?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

mjhuff committed:
refactor(api): Gate grip point calculation behind labware schema version

mjhuff committed:
refactor(api): revert x/y stackingOffsetWithLabware tiprack definition changes

sfoster1 reviewed (approved):
lgtm!

sfoster1 reviewed (approved):
give the pr a nicer name though please!

SyntaxColoring commented on shared-data/labware/schemas/2.json:
```diff
@@ -684,7 +684,7 @@
     "stackingOffsetWithLabware": {
       "type": "object",
-      "description": "Supported labware that this one can be stacked atop, with overlaps between both labware. x=0,y=0,z=0 means the front-left-bottom corner of this labware is at the front-left-top corner of the labware below it. +x, +y, and +z move this labware right, back, and down, respectively.",
+      "description": "Supported labware that this one can be stacked atop, with overlaps between both labware. x and y should always equal 0 to preserve intended behavior. z=0 means the bottom corner of this labware is at the top corner of the labware below it. +z moves this labware down,",
```
```suggestion
      "description": "Supported labware that this one can be stacked atop, with overlaps between both labware. x and y should always equal 0 to preserve intended behavior. z=0 means the bottom of this labware is at the top of the labware below it. +z moves this labware down.",
```

SyntaxColoring reviewed (approved):
Thank you!

mjhuff committed:
typos

Co-authored-by: Max Marrone <max@opentrons.com>

github-actions[bot] committed:
fix(analyses-snapshot-testing): heal api_grip-point-calc-by-lw-schema snapshots (#19442)

This PR was requested on the PR
https://github.com/Opentrons/opentrons/pull/19437

Co-authored-by: mjhuff <64858653+mjhuff@users.noreply.github.com>
